### ISSUE TITLE
Ability to create asymmetric keys

### ISSUE THREAD

**Issue by ChrisJBurns** [fluxcd-community/helm-charts#101]
### Is your feature request related to a problem ?

Currently, when we use the `flux-sync` chart, we are given the option of creating the secret in the desired cluster. When we use Crossplane, we have to provide the secret data that we want the chart to create. But it isn't possible to create those keys in the Crossplane composition and pass them into the `flux-sync` chart. 

### Describe the solution you'd like.

Can we implement functionality that allows this chart to create those keys if specified? We could then use a `kubernetes` provider in Crossplane to retrieve the `public` key to then use it further in the composition when creating deploy keys in Gitlab / Github.

### Describe alternatives you've considered.

I've tried to use the `terraform` Crossplane provider, with the `tls` provider in Terraform to generate those keys.

```yaml
          forProvider:
            source: Inline
            module: |
              // Outputs are written to the connection secret.
              output "private_key" {
                value       = tls_private_key.flux_deploy_keys.private_key_pem
                sensitive = true
              }
              output "public_key" {
                value       = tls_private_key.flux_deploy_keys.public_key_pem
                sensitive = false
              }
              resource "tls_private_key" "flux_deploy_keys" {
                algorithm   = "ECDSA"
                ecdsa_curve = "P256"
              }
```

However, the `private_key` (because it is marked as sensitive) is output into the connection details secret which means I cannot retrieve the value for the `flux-sync` Chart. Pending https://github.com/crossplane/crossplane/issues/2772.

### Additional context.

We are trying to create clusters and install Flux onto those clusters with Crossplane. We can create the cluster, install Flux on it, but we cannot create / give the asymmetric keys to the `flux-sync` Helm Chart that is used to create the `flux-system` secret that the `source-controller` needs in order to clone the repo.

**Comment by stefanprodan** [comment]
There is a `flux create secret git` that generates SSH keys and known host keys which are saved in a Kubernetes Secret. I guess we can use a Helm hook that wraps the CLI like we in the flux2 chart.

**Comment by ChrisJBurns** [comment]
That would be great! The `known_hosts` I knew were also another thing that I needed to figure out how to resolve, but you're right, if there is a `CLI` that we can use in the Chart, that completely resolves the `known_hosts` problem too.

**Comment by ChrisJBurns** [comment]
Have got it working. Will be putting a PR up soon, just need to finalise a few edge cases. Requires a bit more than the `pre-install-job.yaml` as it's writing to `flux-system` namespace, so it needs an SA with some `Role` and `RoleBinding` permissions to write to secrets in the `flux-system` namespace.

**Comment by ChrisJBurns** [comment]
Closing as this has has been implemented in #102
